observability/micrometer/src/main/java/org/wildfly/extension/micrometer/MicrometerSubsystemRegistrar.java

```diff
@@ -0,0 +1,189 @@
+            .setMeasurementUnit(MeasurementUnit.SECONDS)
+            .setAllowExpression(true)
+            .setRestartAllServices()
+            .build();
+
+    static final StringListAttributeDefinition EXPOSED_SUBSYSTEMS =
+            new StringListAttributeDefinition.Builder("exposed-subsystems")
+                    .setDefaultValue(ModelNode.fromJSONString("[\"*\"]"))
+                    .setRequired(false)
+                    .setRestartAllServices()
+                    .build();
+
+    static final AttributeDefinition[] ATTRIBUTES = {
+            EXPOSED_SUBSYSTEMS,
+            ENDPOINT,
+            STEP
+    };
+
+    private WildFlyRegistry wildFlyRegistry;
+    private List<String> exposedSubsystems;
+    private boolean exposeAnySubsystem;
+    private String endpoint;
+    private Long step;
+
+    @Override
+    public ManagementResourceRegistration register(SubsystemRegistration parent,
+                                                   ManagementResourceRegistrationContext context) {
+        ManagementResourceRegistration registration =
+                parent.registerSubsystemModel(ResourceDefinition.builder(ResourceRegistration.of(PATH), RESOLVER).build());
+        ResourceDescriptor descriptor = ResourceDescriptor.builder(RESOLVER)
+                .withRuntimeHandler(ResourceOperationRuntimeHandler.configureService(this))
+                .addCapability(MICROMETER_COLLECTOR_RUNTIME_CAPABILITY)
+                .addAttributes(List.of(ENDPOINT, STEP, EXPOSED_SUBSYSTEMS))
+                .withDeploymentChainContributor(target -> {
+                    target.addDeploymentProcessor(MicrometerConfigurationConstants.NAME, DEPENDENCIES, DEPENDENCIES_MICROMETER,
+                            new MicrometerDependencyProcessor());
+                    target.addDeploymentProcessor(MicrometerConfigurationConstants.NAME, POST_MODULE, POST_MODULE_MICROMETER,
+                            new MicrometerDeploymentProcessor(exposeAnySubsystem, exposedSubsystems, wildFlyRegistry));
+                })
+                .build();
+
+        ManagementResourceRegistrar.of(descriptor).register(registration);
+
+        return registration;
+    }
+
+    @Override
+    public ResourceServiceInstaller configure(OperationContext context, ModelNode model) throws OperationFailedException {
+        exposedSubsystems = MicrometerSubsystemRegistrar.EXPOSED_SUBSYSTEMS.unwrap(context, model);
+        exposeAnySubsystem = exposedSubsystems.remove("*");
+        endpoint = MicrometerSubsystemRegistrar.ENDPOINT.resolveModelAttribute(context, model).asStringOrNull();
+        step = MicrometerSubsystemRegistrar.STEP.resolveModelAttribute(context, model).asLong();
+        wildFlyRegistry = endpoint != null ?
+                new WildFlyOtlpRegistry(new WildFlyMicrometerConfig(endpoint, step)) :
+                new NoOpRegistry();
+
+        try {
+            new JmxMicrometerCollector(wildFlyRegistry).init();
```

Comment:
Doesn't this imply a requirement on "org.wildfly.management.jmx"?

Reply:
Same as above. I'm not sure. It's been that way for years and works, but I can't attest to its correctness.